File: DUImageView.m
//
//  DUImageView.m
//  DictUnifier
//
//  Created by Jjgod Jiang on 3/11/10.
//

#import "DUImageView.h"

@implementation DUImageView

- (void) awakeFromNib
{
    [self registerForDraggedTypes: [NSArray arrayWithObjects: NSPasteboardTypeFileURL, nil]];
}

- (NSDragOperation) draggingEntered: (id < NSDraggingInfo >) sender
{
    NSPasteboard *pboard = [sender draggingPasteboard];
    NSDragOperation opType = NSDragOperationNone;

    if ([[pboard types] containsObject: NSPasteboardTypeFileURL])
        opType = NSDragOperationCopy;

    return opType;
}

- (BOOL) performDragOperation: (id < NSDraggingInfo >) sender
{
    NSPasteboard *pboard = [sender draggingPasteboard];
    BOOL successful = NO;

    if (pboard.pasteboardItems.count <= 1) {
        //直接获取文件路径
        
        NSString *fileURL = [[NSURL URLFromPasteboard:pboard] path];
        [controller startConversion:fileURL];
        successful = NO;
    }
    
    
    return successful;
}

@end

File: DUImageView.py
#
#  DUImageView.py
#  DictUnifier
#
#  Created by Jiang Jiang on 3/7/10.
#  Copyright (c) 2010 Jjgod Jiang. All rights reserved.
#

from objc import YES, NO, IBAction, IBOutlet
from Foundation import *
from AppKit import *

class DUImageView(NSImageView):
    controller = objc.IBOutlet()

    def awakeFromNib(self):
        self.registerForDraggedTypes_([NSFilenamesPboardType])

    def draggingEntered_(self, sender):
        pboard = sender.draggingPasteboard()
        types = pboard.types()
        opType = NSDragOperationNone
        if NSFilenamesPboardType in types:
            opType = NSDragOperationCopy
        return opType

    def performDragOperation_(self,sender):
        pboard = sender.draggingPasteboard()
        successful = False
        if NSFilenamesPboardType in pboard.types():
            files = pboard.propertyListForType_(NSFilenamesPboardType)
            self.controller.startConversion(files[0])
            successful = True
        return successful

